cli/src/commands/git/colocate.rs

```diff
@@ -0,0 +1,293 @@
+// Copyright 2025 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::path::Path;
+
+use crate::cli_util::CommandHelper;
+use crate::cli_util::WorkspaceCommandHelper;
+use crate::command_error::CommandError;
+use crate::command_error::user_error;
+use crate::command_error::user_error_with_message;
+use crate::git_util::is_colocated_git_workspace;
+use crate::ui::Ui;
+
+/// Manage Jujutsu repository colocation with Git
+#[derive(clap::Args, Clone, Debug)]
+pub struct GitColocateArgs {
+    /// Enable co-location (convert into a co-located Jujutsu/Git repository)
+    #[arg(long)]
+    enable: bool,
+
+    /// Disable co-location (convert into a non co-located Jujutsu/Git
+    /// repository)
+    #[arg(long)]
+    disable: bool,
+}
+
+pub fn cmd_git_colocate(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    args: &GitColocateArgs,
+) -> Result<(), CommandError> {
+    let mut workspace_command = command.workspace_helper(ui)?;
+
+    // Validate that both flags aren't used together
+    if args.enable && args.disable {
+        return Err(user_error(
+            "Cannot specify both --enable and --disable flags",
```

Comment:
Prefer a `conflict_with` flag on the option. See https://github.com/jj-vcs/jj/blob/6a1163472ef4f2d85d7b2434402b72cabfcb6a6a/cli/src/commands/metaedit.rs#L82 for an example.